### DISCUSSION TITLE
How to change leader key if inside tmux?

### DISCUSSION THREAD

**Discussion by gipo355** [wezterm/wezterm#4827] [Q&A]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

i3

### WezTerm version

20230424-221505-154dfae9

### Ask your question!

Hello!

For consistency between tmux and wezterm bindings, i'd love if it was possible to change the leader modifier if inside tmux.

At the moment, tmux leader is CTRL+a and wezterm is ALT+a.

I've set up all bindings for wezterm to reflect tmux actions but with the different lead.

What I've tried so far works but only once at startup (it won't dynamically change when i enter/exit tmux).
It basically only stays as CTRL because on init it's wezterm.

```lua
local isTmux = os.getenv("TERM_PROGRAM") == "tmux"
local leaderKey = isTmux and "ALT" or "CTRL"

return {
  	-- leader = { key = "a", mods = "ALT" },
	leader = { key = "a", mods = leaderKey },
}
```

is there a way make it dynamic? When does the config file get called?

thank you very much!

**Comment by gipo355** [comment]
another try i made:

```lua



local function getLeaderKey()
	local isTmux = os.getenv("TERM_PROGRAM") == "tmux"
	if isTmux then
		return "ALT"
	else
		return "CTRL"
	end
end

leader = { key = "a", mods = getLeaderKey() },

-- set keybind below:
		{
			key = "r",
			mods = "LEADER",
			action = wezterm.action.ReloadConfiguration,
		},

```

even forcing the reload doesn't work

**Comment by wez** [comment]
I'd recommend configuring tmux to emit a user var (eg: `IS_TMUX`) when it starts (or do that in a shell alias), and clear it when it exits. Then you can take action on that in your wezterm config.

Take a look at:
* https://wezfurlong.org/wezterm/recipes/passing-data.html (for a discussion on user vars)
* https://github.com/wez/wezterm/discussions/3648 (which is about nvim, but the concept is the same)

**Comment by gipo355** [comment]
thank you wez for the direction!

**Reply by govardha** [reply]
Hello

did you get this to work?  If so can you please share your function ?

thanks 
G
